bin/live/pycbc_live_single_trigger_fits

```diff
@@ -0,0 +1,377 @@
+
+if args.verbose:
+    log_level = logging.DEBUG
+else:
+    log_level = logging.WARN
+
+logging.basicConfig(format='%(asctime)s : %(message)s', level=log_level)
+
+files = [f for f in os.listdir(os.path.join(args.top_directory, args.analysis_date))
+         if args.file_identifier in f]
+
+logging.info("{} files found".format(len(files)))
+
+args.cut_params.append('template_duration')
+args.cut_inequalities.append("<")
+args.cut_thresholds.append(args.duration_bin_end)
+
+args.cut_params.append('template_duration')
+args.cut_inequalities.append(">")
+args.cut_thresholds.append(args.duration_bin_start)
+
+cut_inequalities = np.array(args.cut_inequalities)
+cut_params = np.array(args.cut_params)
+cut_thresholds = np.array(args.cut_thresholds)
+
+if args.duration_bin_spacing == 'log':
+    bincreator = bin_utils.LogarithmicBins
+elif args.duration_bin_spacing == 'linear':
+    bincreator = bin_utils.LinearBins
+
+tbins = bincreator(args.duration_bin_start,
+                   args.duration_bin_end,
+                   args.num_duration_bins)
+
+
+# newsnr needs special treatment as they are not
+# in the saved files. Others are either in the saved files or can be
+# calculated in the get_param function
+
+cutting_newsnr = False
+if 'new_snr' in cut_params:
+    cutting_newsnr = True
+    nsidx = np.nonzero(cut_params == 'new_snr')[0]
+    # To save on chisq & newsnr calculation costs, and memory use, cut on
+    # snr which, by definition, is greater than or equal to newsnr
+    cut_params[nsidx] = 'snr'
+    nsineq = cut_inequalities[nsidx]
+    nsthresh = cut_thresholds[nsidx]
+
+cut_params = list(cut_params)
+cut_inequalities = list(cut_inequalities)
+cut_thresholds = list(cut_thresholds)
+
+# The following datasets are required for calculation of cut parameters
+required_datasets = ['snr', 'chisq', 'chisq_dof', 'template_duration',
+                     'mass1', 'mass2', 'spin1z', 'spin2z']
+required_datasets += cut_params
+required_datasets = list(set(required_datasets))
+
+# Make some strings for logging /saving setting for attributes
+zip_inequs = zip(cut_params, cut_inequalities, cut_thresholds)
+ineq_str = ''
+for p, i, v in zip_inequs:
+    ineq_str += '{} {} {}, '.format(p, i, v)
+ineq_str = ineq_str[:-2]
+
+all_zip_inequs = zip(args.cut_params, args.cut_inequalities,
+                     args.cut_thresholds)
+ineq_str_all = ''
+for p, i, v in all_zip_inequs:
+    ineq_str_all += '{} {} {}, '.format(p, i, v)
+ineq_str_all= ineq_str_all[:-2]
+logging.info('Criteria set up as ' + ineq_str_all)
+
+# Also calculate live time so that this fitting can be used in rate estimation
+# Live time is not immediately obvious - get an approximation with 8 second
+# granularity by adding 8 seconds per 'valid' file
+
+live_time = {ifo: 0 for ifo in args.ifos}
+
+#store as a DictArray - nice as it has the .select() method
+events_dict = {k : [] for k in required_datasets}
+logging.info("Getting events from which meet criteria: {}".format(ineq_str))
```

Comment:
Logging methods do their own formatting, and are generally optimized to skip formatting operations if no logging handlers are configured. I'd suggest:

```python
logging.info("Getting events from which meet criteria: %s", ineq_str)
```

Reply:
That's good to know! Implemented here but I will look in other codes to see if that will improve efficiency